File: MergeConflictAnalysis/src/edu/oregonstate/mergeproblem/mergeconflictanalysis/processors/CompositeProcessor.java
package edu.oregonstate.mergeproblem.mergeconflictanalysis.processors;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

import edu.oregonstate.mergeproblem.mergeconflictanalysis.CommitStatus;

public class CompositeProcessor implements FileProcessor {
	
	private List<FileProcessor> processors = new ArrayList<FileProcessor>();
	
	public void addProcessor(FileProcessor processor) {
		processors.add(processor);
	}

	@Override
	public String getHeader() {
		return processors.stream()
				.map((processor) -> processor.getHeader())
				.collect(Collectors.joining(","));
	}

	@Override
	public String getDataForMerge(CommitStatus status, String fileName) {
		return processors.stream()
				.map((processor) -> processor.getDataForMerge(status, fileName))
				.collect(Collectors.joining(","));
	}

}

File: MergeConflictAnalysis/src/main/scala/edu/oregonstate/mergeproblem/mergeconflictanalysis/processors/CompositeProcessor.scala
package edu.oregonstate.mergeproblem.mergeconflictanalysis.processors

import edu.oregonstate.mergeproblem.mergeconflictanalysis.file.CommitStatus

import scala.collection.mutable.ListBuffer

class CompositeProcessor extends FileProcessor {
	private val processors  = new ListBuffer[FileProcessor]

	def addProcessor(processor: FileProcessor) =
		processors :+ processor

	def getHeader: String =
		processors map { _.getHeader } mkString ","

	def getData(status: CommitStatus, fileName: String): String =
		processors map { _.getData(status, fileName) } mkString ","
}